core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -16,18 +16,15 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
```

Comment:
why this big refactor is needed for `Limit maximum size of dynamic filter collected by coordinator`?

Reply:
The refactor is needed to merge dynamic filters incrementally. Enforcing maximum filter size before domains are merged together may result in hitting the limit prematurely